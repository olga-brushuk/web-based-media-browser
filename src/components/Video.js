import React, { Component } from 'react';

class Video extends Component {

  render() {
    // depending on the source platform of the track generate different track id
    const trackLinkIdentifier = this.props.trackId;
    let trackId;

    if (trackLinkIdentifier.startsWith('/yt/')) { // Youtube source platform
      trackId = trackLinkIdentifier.replace('/yt/', ''); // get the id only

      return (
        <div className="track__video">
          <iframe type="text/html" width="640" height="360" frameBorder="no" title={this.props.trackName} 
          src={`https://www.youtube.com/embed/${trackId}`} allowFullScreen></iframe>
        </div>
      );
    } 
    else if (trackLinkIdentifier.startsWith('/sc/')) { // SoundCloud source platform
      // get the id only (maybe this can be improved or implemented in a different way, doesn't seem reliable)
      trackId = trackLinkIdentifier.split('tracks/')[1].slice(0, -7);
      
      return (
        <div className="track__video">
          <iframe width="100%" height="166" scrolling="no" frameBorder="no" title={this.props.trackName}
          src={`https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/${trackId}`}></iframe>
        </div>
      );
    } else { // fallback for other providers as it was not implemented yet
      return (
        <p>The track playback is not available</p>
      );
    }
  }
}

export default Video;