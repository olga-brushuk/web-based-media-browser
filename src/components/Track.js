import React, { Component } from 'react';
import Video from './Video';

class Track extends Component {
  render() {
    return (
      <div className="track">
        <div className="track__thumb">
          {/* Use image thumbnail and play video via API for each of the source platforms
          <img src={this.props.trackThumb} alt={this.props.trackName} />
          <div className="track__play"></div> */}
          <Video trackId={this.props.trackEId} trackName={this.props.trackName} />
        </div>
        <div className="track__info">
          <h2>{this.props.trackName}</h2>
          <a href={`https://openwhyd.org/u/${this.props.userId}`} className="track__user-info">
            <div className="track__user-avatar" 
            style={{backgroundImage:`url(https://openwhyd.org/img/user/${this.props.userId})`}}></div>
            <div className="track__username">{this.props.userName}</div>
          </a>
        </div>
      </div>
    );
  }
}

export default Track;