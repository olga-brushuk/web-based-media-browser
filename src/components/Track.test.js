import React from 'react';

import {configure, shallow} from 'enzyme';

import Track from './Track';
import Video from './Video';

let wrapped = shallow(<Track 
    trackName="Brooks - If Only I Could"
    userName="Sabrina"
    userId="50bcaf297e91c862b2a91d7c"
    trackEId="/yt/U7KQ-8I_jYI" 
/>);

describe('<Track />', () => {
    it('render correctly <Track /> component', () => {  
        expect(wrapped).toMatchSnapshot();
    });
    it('should render one <Video /> element in <Track /> component', () => {
       expect(wrapped.find(Video)).toHaveLength(1);
    });
    it('should render track title correctly in <Track /> component', () => {
        expect(wrapped.find('.track__info > h2').text()).toEqual('Brooks - If Only I Could');
    });
    it('should render track username correctly in <Track /> component', () => {
        expect(wrapped.find('.track__username').text()).toEqual('Sabrina');
    });
    it('should render track user id correctly in <Track /> component', () => {
        expect(wrapped.find('.track__user-info').props().href).toEqual('https://openwhyd.org/u/50bcaf297e91c862b2a91d7c');
    });
});