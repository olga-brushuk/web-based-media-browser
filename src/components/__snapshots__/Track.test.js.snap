// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Track /> render correctly <Track /> component 1`] = `
<div
  className="track"
>
  <div
    className="track__thumb"
  >
    <Video
      trackId="/yt/U7KQ-8I_jYI"
      trackName="Brooks - If Only I Could"
    />
  </div>
  <div
    className="track__info"
  >
    <h2>
      Brooks - If Only I Could
    </h2>
    <a
      className="track__user-info"
      href="https://openwhyd.org/u/50bcaf297e91c862b2a91d7c"
    >
      <div
        className="track__user-avatar"
        style={
          Object {
            "backgroundImage": "url(https://openwhyd.org/img/user/50bcaf297e91c862b2a91d7c)",
          }
        }
      />
      <div
        className="track__username"
      >
        Sabrina
      </div>
    </a>
  </div>
</div>
`;
